app/demo-loop/LDemoIO.cc

```diff
@@ -129,6 +149,20 @@ void from_json(nlohmann::json const& j, LDemoArgs& v)
     {
         j.at("mctruth_filename").get_to(v.mctruth_filename);
     }
+    if (j.contains("mctruth_filter"))
+    {
+        auto const& jfilter = j.at("mctruth_filter");
+        get_optional(jfilter, "event_id", v.mctruth_filter.event_id);
+        get_optional(jfilter, "track_id", v.mctruth_filter.track_id);
+        get_optional(jfilter, "parent_id", v.mctruth_filter.parent_id);
+
+        if (v.mctruth_filter)
+        {
+            CELER_VALIDATE(!v.mctruth_filename.empty(),
+                           << "A valid MC truth filter can only be used with "
+                              "a valid ROOT MC truth output");
```

Comment:
The assertion message should say what's wrong as the primary message, not be a statement about what conditions are correct
```suggestion
                           << "missing 'mctruth_filename' when 'mctruth_filter' was specified");
```